### PR TITLE
Sync only specific instance id

### PR DESCRIPTION
**Pull request by alenkacz**

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala [hunk 1]
```diff
@@ -155,7 +155,7 @@ class TaskReplaceActor(
 
   def reconcileAlreadyStartedInstances(): Unit = {
     logger.info(s"reconcile: found ${instancesAlreadyStarted.size} already started instances " +
-      s"and ${oldInstanceIds.size} old instances: ${currentRunningInstances.map{ i => i.instanceId -> i.state.condition }}")
+      s"and ${oldInstanceIds.size} old instances: ${currentInstances.map{ i => i.instanceId -> i.state.condition }}")
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala]: How about logging the goal and the condition?

**Comment by jeschkies** [comment]
So I see

```
old version 2018-07-24T14:21:46.973Z to
new version 2018-07-24T14:21:48.110Z

instance [app-with-backoff-delay-is-reset-on-conf-changes.instance-e6588f8d-8f4c-11e8-89bc-02429e0d1256]:{condition: Scheduled, goal: Running, version: 2018-07-24T14:21:46.973Z, reservation: None, 
instance [app-with-backoff-delay-is-reset-on-conf-changes.instance-e683972e-8f4c-11e8-89bc-02429e0d1256]:{condition: Scheduled, goal: Running, version: 2018-07-24T14:21:48.110Z, reservation: None
instance [app-with-backoff-delay-is-reset-on-conf-changes.instance-eb565d7e-8f3e-11e8-ba52-024295b3bdfa]:{condition: Scheduled, goal: Running, version: 2018-07-24T12:41:42.978Z, reservation: None}
```

We have three apps with different versions each :/.

**Comment by jeschkies** [comment]
I think I've found another issue in `InstanceUpdater:48`:

```scala
private def shouldBeExpunged(instance: Instance): Boolean =
  instance.tasksMap.values.forall(_.isTerminal) && instance.state.goal != Goal.Stopped
```

this should be

```scala
private def shouldBeExpunged(instance: Instance): Boolean =
  instance.tasksMap.values.forall(_.isTerminal) && instance.state.goal == Goal.Decommissioned
```

A scheduled instances has no tasks so `instance.tasksMap.values.forall(_.isTerminal)` evaluates to `true`. See https://scastie.scala-lang.org/KFpL3wLqSCi4ebO00FF6DA.

**Comment by jeschkies** [comment]
I just saw that the instance is updated with the task. It should be `goal == Goal.Decommissioned` Nonetheless.

**Comment by jeschkies** [comment]
I've just realized that we did the expunge to replace failed instances. Let's discuss this tomorrow.

**Comment by alenkacz** [comment]
well we can't do `instance.state.goal == Goal.Decommissioned` because we don't decommission ephemeral instances :/

**Comment by jeschkies** [comment]
> well we can't do instance.state.goal == Goal.Decommissioned because we don't decommission ephemeral instances :/

We do decommission ephemeral instances. In fact, we always decommission them instead of reusing the same instance on a restart.

**Comment by alenkacz** [comment]
@jeschkies Yeah, but we don't set the goal to decommissioned.

**Pull request merged**